core/trino-spi/src/main/java/io/trino/spi/type/RowType.java

```diff
@@ -136,6 +137,17 @@ public static RowType anonymous(List<Type> types)
         return new RowType(makeSignature(fields), fields);
     }
 
+    public static RowType withDefaultFieldNames(List<Type> types)
```

Comment:
You should not modify production code to satisfy the needs of the tests. Especially in SPI.
Move this to some testing class